src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs

```diff
@@ -261,284 +260,48 @@ internal static ImmutableArray<IDocumentSnapshot> GetImportsCore(IProjectSnapsho
         return imports.ToImmutable();
     }
 
-    // Internal, because we are temporarily sharing code with CohostDocumentSnapshot
-    internal class ComputedStateTracker
+    internal static async Task<RazorCodeDocument> GenerateCodeDocumentAsync(ImmutableArray<TagHelperDescriptor> tagHelpers, RazorProjectEngine projectEngine, IDocumentSnapshot document, ImmutableArray<ImportItem> imports, bool forceRuntimeCodeGeneration)
```

Comment:
I moved stuff around but I'm not totally happy with how it ended up. Maybe it's time to move these into a static class on their own?